Tests/SwiftStdlibTests/OptionalExtensionsTests.swift

```diff
@@ -105,4 +105,39 @@ final class OptionalExtensionsTests: XCTestCase {
 
     }
 
+    func testComparable() {
+        let minusOne: Int? = -1
+        let zero: Int? = 0
+        let one: Int? = 1
+
+        // Int?.some vs Int
+        XCTAssert(minusOne < 0)
+        XCTAssert(zero < 1)
+        XCTAssert(one < 2)
+        XCTAssert(-2 < minusOne)
+        XCTAssert(-1 < zero)
+        XCTAssert(0 < one)
+
+        // Int?.none vs Int
+        XCTAssert(nil < -1)
+        XCTAssert(nil < 0)
+        XCTAssert(nil < 1)
+        XCTAssertFalse(-1 < nil)
```

Comment:
This really feels weird to me, I mean for correctness saying that an integer or any value is greater or lesser than nil, because, at least for me it can't be true or false because nil is not actually a value. Maybe that was part of the reason for this not to be on the stdlib, but I'm not sure, is just a feeling that I have ... so let me know your thoughts on that :))

Reply:
I see it as `nil` being the absence of a value. Maybe it reads better if I use `Int?.none` instead of `nil`.
```Swift
XCTAssert(Int?.none < 0)
```

Reply:
The issue is not how it reads better, but actually with semantics and correctness of comparing something to nil. Even as you mention, if we assume nothing is less than nil like you mention python does.

Reply:
This is the tricky bit about the PR, why I wanted to get more opinions about it. Because yeah, semantically it doesn't make sense to compare to `nil` as a non-value, but `nil` as `Int?.none` can be semantically true. @SD10 @BennX @omaralbeik can I get your opinions about this too, please?